src/acceptor.rs

```diff
@@ -0,0 +1,712 @@
+    /// Create a new rustls_acceptor.
+    ///
+    /// Parameters:
+    ///
+    /// out_acceptor: An output parameter. The pointed-to pointer will be set
+    ///   to a new rustls_acceptor on success.
+    ///
+    /// Returns:
+    ///
+    /// - RUSTLS_RESULT_OK: Success. *rustls_acceptor has been written to.
+    /// - Other rustls_result: Error. *rustls_acceptor has not been written to.
+    ///
+    /// Memory and lifetimes:
+    ///
+    /// On success, caller owns the pointed-to memory and must eventually free
+    /// it with rustls_acceptor_free().
+    #[no_mangle]
+    pub extern "C" fn rustls_acceptor_new(
+        out_acceptor: *mut *mut rustls_acceptor,
+    ) -> rustls_result {
+        ffi_panic_boundary! {
+            match Acceptor::new() {
+                Ok(acceptor) => {
+                    BoxCastPtr::set_mut_ptr(out_acceptor, acceptor);
+                    rustls_result::Ok
+                },
+                Err(e) => map_error(e),
+            }
+        }
+    }
+
+    /// Free a rustls_acceptor.
+    ///
+    /// Parameters:
+    ///
+    /// acceptor: The rustls_acceptor to free.
+    ///
+    /// Calling with NULL is fine. Must not be called twice with the same value.
+    #[no_mangle]
+    pub extern "C" fn rustls_acceptor_free(acceptor: *mut rustls_acceptor) {
+        ffi_panic_boundary! {
+            BoxCastPtr::to_box(acceptor);
+        }
+    }
+
+    /// Check if this rustls_acceptor wants additional TLS bytes read into it.
+    /// If this returns true, you should call rustls_acceptor_read_tls().
+    ///
+    /// Parameters:
+    ///
+    /// acceptor: The rustls_acceptor to check.
+    ///
+    /// Returns:
+    ///
+    /// True if and only iff this rustls_acceptor wants additional TLS bytes
+    /// (and is non-NULL).
+    #[no_mangle]
+    pub extern "C" fn rustls_acceptor_wants_read(acceptor: *const rustls_acceptor) -> bool {
+        ffi_panic_boundary! {
+            let acceptor = try_ref_from_ptr!(acceptor);
+            acceptor.wants_read()
+        }
+    }
+
+    /// Read some TLS bytes from the network into internal buffers. The actual network
+    /// I/O is performed by `callback`, which you provide. Rustls will invoke your
+    /// callback with a suitable buffer to store the read bytes into. You don't have
+    /// to fill it up, just fill with as many bytes as you get in one syscall.
+    ///
+    /// Parameters:
+    ///
+    /// acceptor: The rustls_acceptor to read bytes into.
+    /// callback: A function that will perform the actual network I/O.
+    /// userdata: An opaque parameter to be passed directly to `callback`.
+    ///   Note: this is distinct from the `userdata` parameter set with
+    ///   `rustls_connection_set_userdata`.
+    /// out_n: An output parameter. On success, this will be written with
+    ///   the number of bytes `callback` said that it wrote.
+    ///
+    /// Returns:
+    ///
+    /// - 0: Success. You should call rustls_acceptor_accept() next.
```

Comment:
Maybe also backtick-quote things like `rustls_acceptor_accept()` and `callback` and `rustls_read_callback` (below)?